### PR TITLE
ECM-216 fixed

### PR DESCRIPTION
**Pull request by gauravchawla73463**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ECM-216

### Change description ###

In a case where the Respondent is struck out, their Representative's details should not appear in letters

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/DocumentHelperTest.java [hunk 1]
```diff
@@ -164,7 +162,84 @@ public void buildDocumentContent1() {
...
+                + "\"Case_No\":\"123456\",\n"
+                + "}\n"
+                + "}\n";
+        caseDetails2.getCaseData().getRepCollection().get(0).getValue().setRespRepName("Antonio Vazquez");
+        assertEquals(expected, DocumentHelper.buildDocumentContent(caseDetails2.getCaseData(), "",
+                userDetails, MANCHESTER_CASE_TYPE_ID, venueAddressInputStream,
+                caseDetails2.getCaseData().getCorrespondenceType(),
+                caseDetails2.getCaseData().getCorrespondenceScotType(), null,
+                null).toString());
+        caseDetails2.getCaseData().getRepCollection().get(0).getValue().setRespRepName("RepresentativeNameRespondent");
```
Comment (jthmcts) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/DocumentHelperTest.java:238]: Does this line serve any purpose?

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/DocumentHelperTest.java [hunk 2]
```diff
@@ -233,11 +308,14 @@ public void buildDocumentContent2() {
...
                 + "}\n"
                 + "}\n";
+        caseDetails2.getCaseData().getRespondentCollection().get(0).getValue().setResponseStruckOut(NO);
+        caseDetails2.getCaseData().getRepCollection().get(0).getValue().setRespRepName("Antonio Vazquez");
         assertEquals(expected, DocumentHelper.buildDocumentContent(caseDetails2.getCaseData(), "",
                 userDetails, MANCHESTER_CASE_TYPE_ID, venueAddressInputStream,
                 caseDetails2.getCaseData().getCorrespondenceType(),
                 caseDetails2.getCaseData().getCorrespondenceScotType(), null,
                 null).toString());
+        caseDetails2.getCaseData().getRepCollection().get(0).getValue().setRespRepName("RepresentativeNameRespondent");
```
Comment (jthmcts) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/DocumentHelperTest.java:318]: Does this line serve any purpose?

**Review by jthmcts — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/DocumentHelperTest.java (hunk 2 above)
Comment (gauravchawla73463) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/DocumentHelperTest.java:318]: Unfortunately other tests fail, if this line is not there. We have to reset the RespRepName to the initial value as it was in CaseDetailsTest2.json.

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/DocumentHelperTest.java (hunk 1 above)
Comment (gauravchawla73463) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/DocumentHelperTest.java:238]: Unfortunately other tests fail, if this line is not there. We have to reset the RespRepName to the initial value as it was in CaseDetailsTest2.json.

**Review by jthmcts — APPROVED** [review]

**Pull request merged**